tools/kprint/main.cpp

```diff
@@ -1,33 +1,36 @@
 #include "kllvm/ast/AST.h"
 #include "kllvm/printer/printer.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/WithColor.h"
 
 #include <iostream>
 #include <string>
 #include <unistd.h>
 
 using namespace kllvm;
+using namespace llvm;
+
+cl::opt<std::string> DefinitionFilename(
+    cl::Positional, cl::desc("<definition.kore>"), cl::Required);
+
+cl::opt<std::string>
+    PatternFilename(cl::Positional, cl::desc("<pattern.kore>"), cl::Required);
+
+cl::opt<std::string> FilterSubst(cl::Positional, cl::desc("[true|false]"));
+
+cl::opt<std::string> ArgColor(cl::Positional, cl::desc("[true|false|auto]"));
 
 int main(int argc, char **argv) {
-  if (argc != 3 && argc != 4 && argc != 5) {
-    std::cerr
-        << "usage: " << argv[0]
-        << " <definition.kore> <pattern.kore> [true|false|auto] [true|false]"
-        << std::endl;
-  }
-
-  bool hasColor, filterSubst;
-  if (argc >= 4) {
-    std::string arg = argv[3];
-    hasColor = arg == "true" || (arg == "auto" && isatty(1));
-  } else {
-    hasColor = isatty(1);
-  }
-  if (argc >= 5) {
-    std::string arg = argv[4];
-    filterSubst = arg == "true";
-  } else {
-    filterSubst = true;
-  }
-
-  printKORE(std::cout, argv[1], argv[2], hasColor, filterSubst);
+  cl::PrintOptionValues();
+  StringMap<cl::Option *> &Map = cl::getRegisteredOptions();
+  Map["color"]->setHiddenFlag(cl::Hidden);
```

Comment:
_Very_ nitpicky, but I have a preference for almost-always-auto and `snake_case` for new code. Working out a set of C++ guidelines and style choices is on my list of things to do as part of the backend documentation updates.

Reply:
In this file, we have 2 conventions:
- upper camel case for `cl::opt` variables; and
- camelCase for the other variables

Should I modify all variables for the `snake_case`?

Reply:
Good point - option variables I do prefer to keep as UpperCamelCase, but locals as snake_case